doc/architecture/timestamps_overview.adoc

```diff
@@ -0,0 +1,67 @@
+ifndef::include-only-once[]
+:root-path: ../
+include::{root-path}_config.adoc[]
+endif::[]
+= Timestamps
+
+== Simulation time
+
+`Simulation time` refers to the timestamp within a simulation environment.
+
+**Definition of zero time of simulation time**
+- Zero time is arbitrary but must be identical for all frames and corresponding messages (probably within one simulation).
+- Zero time does not need to coincide with the unix epoch.
+- It is recommended to use the start time of the simulation as zero time.
+
+**SensorData-specific timestamp**
+- Point in time that the sensor data message becomes available to the rest of the system (i.e. the driving functions).
```

Comment:
I think the example and the term "becoming available" is a bit misleading. Becoming available to the driving functions cannot really be the same timestamp which excludes but latency, right?

As an idea:
```suggestion
- Represents the exact point in time when the sensor data is published to the rest of the system.
- Accounts for any internal processing latency within the sensor.
- Excludes delays caused by bus communication (e.g., latency after sensor output).

```

Reply:
> I think the example and the term "becoming available" is a bit misleading. Becoming available to the driving functions cannot really be the same timestamp which excludes but latency, right?

The specification currently does not define what becoming available means. This allows for use cases where the whole delay (including any communication delays) is modeled/included in the sensor model (which is useful for many simpler use cases, especially where there is no contention on the communication channel, and/or exact bus modelling has no perceived benefit). It also allows for use cases where the timestamp reflects the sending time, and latency/communication effects are handled outside the sensor model.

Not saying that this situation is ideal, i.e. it would likely be good to have some machine-readable indication which it is, but there is little we can do in pure clarification/maintenance releases.

Reply:
Okay, understood that we cannot change that much. But how does the following:

>  This allows for use cases where the whole delay (including any communication delays) is modeled/included in the sensor model 

fit with the current specification of:

> - Excludes delays caused by bus communication (e.g., latency after sensor output).

or from the current SensorData timestamp doc:

> Latencies of bus communications, etc., that occur after the sensor output have to be applied on top of this, if needed.

Reply:
> That is not actually part of the current OSI specification, though, is it?

That is part of this PR. 3 lines below.


>  however the if needed leaves the door open for use case 1, i.e. if a simplified communication delay is already included in the sensor model, obviously adding bus latency on top is not needed.

I do not see how the "if needed" allows to additional include these in the timestamp field? I understand it that way: "If you need communication latency (like bus latency) you need to add that this latency in the consumer of your  SensorData.timestamp"

> but on the other hand people will do what they want to anyway

Is that not the motivation to write a **good** standard?

> details at this level really have to be documented and coordinated between parties in any case

You are totally right! But I just started this thread about the "becoming available" not being accurate with the following latency mentioning in the text on this added documentation. I  do not want to talk about different forms of latency and where they should be included (nonetheless, I still did that in the beginning of this particular comment ;) ).

Reply:
Strictly excluding all communication delays is not the same as saying that communication delays **after sensor output** are not included or should be added on top. The changed wording would disallow modelling delays within a sensor model whereas the old wording does not. I have changed the original wording in this PR to make it more technical and precise and less exemplary, but maybe I've made it more misleading and it shouldn't be changed because minor wording changes result in major content changes.

To me, "becoming available" (together with the rest of the definition) means that from that point the message is potentially accessible by other (simulation) components without specifying what the timestamp exactly means reality-wise.

So currently the timestamp describes the point in time at which the sensor model component (=sender) has released the message to the outside system regardless of what has happened internally, if I understand that correctly. "Simulated publish time" could also be contained in a sensor model but the message itself becomes available to the rest of the (simulation) system later. The only restriction we can extract from the current specification is that the timestamp must not contain delays that occur after the sensor model output. Did I get this right?

I guess the question is whether it is necessary and/or possible to clarify the actual content of the definition without changing it? Maybe we should just collect issues we see with the current state for future minor/major releases (like adding an indication for the two valid use cases as Pierre suggested)?

_(Side note, because I stumbled across this while thinking and researching about the different terms: Although it should be clear from the context, 'send time' might be a more appropriate term than 'sending time', as the latter is often understood as the duration of the transmission process rather than a single point in time.)_